tests/MqttClientTest.cpp

```diff
@@ -349,3 +357,108 @@ static int s_TestIoTMqtt311ConnectWithSigningCustomAuth(Aws::Crt::Allocator *all
     return AWS_OP_SUCCESS;
 }
 AWS_TEST_CASE(IoTMqtt311ConnectWithSigningCustomAuth, s_TestIoTMqtt311ConnectWithSigningCustomAuth)
+
+/*
+ * PKCS12 connect for MQTT311
+ */
+static int s_TestIoTMqtt311ConnectWithPKCS12(Aws::Crt::Allocator *allocator, void *)
+{
+    struct aws_string *endpoint = NULL;
+    struct aws_string *pkcs12_key = NULL;
+    struct aws_string *pkcs12_password = NULL;
+    struct aws_string *codebuild_buildID = NULL;
+
+    int error = aws_get_environment_value(allocator, s_mqtt5_test_envName_iot_hostname, &endpoint);
+    error |= aws_get_environment_value(allocator, s_mqtt311_test_envName_iot_pkcs12_key, &pkcs12_key);
+    error |= aws_get_environment_value(allocator, s_mqtt311_test_envName_iot_pkcs12_key_password, &pkcs12_password);
+    error |= aws_get_environment_value(allocator, s_mqtt311_test_envName_codebuild, &codebuild_buildID);
+
+    bool isEveryEnvVarSet = (endpoint && pkcs12_key && pkcs12_password);
+    if (isEveryEnvVarSet == true)
+    {
+        isEveryEnvVarSet =
+            (aws_string_is_valid(endpoint) && aws_string_is_valid(pkcs12_key) && aws_string_is_valid(pkcs12_password));
+    }
+    if (error != AWS_OP_SUCCESS || isEveryEnvVarSet == false)
+    {
+        printf("Environment Variables are not set for the test, skip the test");
+        aws_string_destroy(endpoint);
+        aws_string_destroy(pkcs12_key);
+        aws_string_destroy(pkcs12_password);
+
+        // If in Codebuild, return as a 'success' even though it is a skip
+        if (codebuild_buildID && aws_string_is_valid(codebuild_buildID))
+        {
+            aws_string_destroy(codebuild_buildID);
+            return AWS_OP_SUCCESS;
+        }
+        aws_string_destroy(codebuild_buildID);
```

Comment:
Note: This is an unfortunate side effect of Codebuild counting a skipped test as a failure. This is probably something we'll need to look into further later, and I'll probably do so when we get to the C++ CRT in the unit-test/sample split work, but for now, I'm just checking for a Codebuild-specific environment variable and "passing" the test if it exists.